modules/nextflow/src/main/groovy/nextflow/executor/PbsProExecutor.groovy

```diff
@@ -44,12 +46,31 @@ class PbsProExecutor extends PbsExecutor {
     @Override
     protected List<String> getDirectives(TaskRun task, List<String> result ) {
```

Comment:
This ended up being more complicated than I expected. I tried to write my own version of `getDirectives` based on your changes:
```groovy
    protected List<String> getDirectives(TaskRun task, List<String> result ) {
        assert result != null

        // extract resource string from cluster options
        def clusterOptionsList = task.config.getClusterOptionsAsList()
        def resIndex = clusterOptionsList.indexOf('-l')
        def res = ''

        if( resIndex != -1 ) {
            res = clusterOptionsList.removeAt(resIndex + 1)
            clusterOptionsList.removeAt(resIndex)
        }

        // parse resource string into a map
        Map<String, String> resMap = [:]
        for ( String entry : res.split(':') ) {
            String[] tokens = entry.split('=')
            resMap.put(tokens[0], tokens[1])
        }

        // append cpus and memory to resource map
        if( task.config.hasCpus() || task.config.memory ) {
            resMap['ncpus'] = task.config.getCpus().toString()
        }
        if( task.config.memory ) {
            // https://www.osc.edu/documentation/knowledge_base/out_of_memory_oom_or_excessive_memory_usage
            resMap['mem'] = "${task.config.getMemory().getMega()}mb".toString()
        }

        // append cluster options
        for( entry : clusterOptionsList ) {
            result << entry
        }

        // append standard options
        result << '-N' << getJobNameFor(task)
        result << '-o' << quote(task.workDir.resolve(TaskRun.CMD_LOG))
        result << '-j' << 'oe'

        // append queue name
        if( task.config.queue ) {
            result << '-q' << task.config.queue.toString()
        }

        // append final resource list
        final resSelect = resMap['select'] ?: 1
        final resList = resMap
            .findAll(key, value -> key != 'select')
            .collect(key, value -> "${key}=${value}")

        if( !resList.isEmpty() ) {
            result << '-l' << "select=${resSelect}:${resList.join(':')}".toString()
        }

        // max task duration
        if( task.config.time ) {
            final duration = task.config.getTime()
            result << "-l" << "walltime=${duration.format('HH:mm:ss')}".toString()
        }

        return result
    }
```